tests/integration/memcached/test_add_servers.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*DESCRIPTION
+The agent should report instance metrics when multiple servers are
+added at once
+*/
+
+/*SKIPIF
+<?php require('skipif.inc'); ?>
+*/
+
+/*INI
+*/
+
+/*EXPECT_METRICS_EXIST
+Datastore/instance/Memcached/host1/1
+Datastore/instance/Memcached/host2/2
+Datastore/instance/Memcached/host3/11211
+*/
+
+require_once(realpath (dirname ( __FILE__ )) . '/../../include/helpers.php');
+require_once(realpath (dirname ( __FILE__ )) . '/../../include/tap.php');
+require_once(realpath (dirname ( __FILE__ )) . '/memcache.inc');
+
+$memcached = new Memcached();
+$memcached->addServers(array(
+                       array("host1", 1),
+                       array("host2", 2),
+                       array("host3", 11211)));
+$memcached->addServers(array());
```

Comment:
Can following tests be added as well:
- with non-array
- with an array of array that only has one element - host name a as a string
- with an array of array that only has one element - host name as a number
- with an array of array that has two elements - different combos
- with an array of array that has more than 3 elements

Reply:
I have added some additional tests (and more robust type handling in the code) here [e141481](https://github.com/newrelic/newrelic-php-agent/pull/958/commits/e141481475416099da72d1cab770d748844640e8)

This cannot be tested with a non-array, as that crashes PHP

Reply:
> This cannot be tested with a non-array, as that crashes PHP

Could you add commented test case for that with your explanation, please?